OpenMage/magento-lts
# Used transliterator_transliterate to generate url_key

### Description

When a product or a category url key use special characters, they are substituted with **strstr**.
With this PR, **transliteration** is used (with [PHP INTL](https://www.php.net/transliterator_transliterate)).

Tested with the following locales: fr-FR, es-ES, de-DE, it-IT, pt-PT, nl-NL, pl-PL, cs-CZ, sk-SK, ru-RU, uk-UA.
OpenMage 20.x / PHP 7.4.6 and 8.0.x and 8.2.x

### Manual testing scenarios

- Set default locale to _French_.
- Set first store locale to _Russian_.
- Set second store locale to _French_.
- Create a product and category with the following name: _Массажный Аппарат для Похудения A_
- Set the product and category name on first store to:  _Массажный Аппарат для Похудения B_
- Set the product and category name on second store to:  _Массажный Аппарат для Похудения C_

Run:
```sql
DELETE FROM catalog_category_entity_varchar WHERE attribute_id IN (
    SELECT attribute_id FROM eav_attribute WHERE attribute_code LIKE 'url_key');
DELETE FROM catalog_product_entity_varchar WHERE attribute_id IN (
    SELECT attribute_id FROM eav_attribute WHERE attribute_code LIKE 'url_key');
```
Run:
```
rm -rf var/cache/ ; php shell/indexer.php reindexall ; rm -rf var/cache/
```

Check url keys generated:
- all store views: `massazhnyj-apparat-dlja-pohudenija-b`
- first store: `massazhnyj-apparat-dlja-pohudenija-b`
- second store: `massazhnyj-apparat-dlja-pohudenija-b`

That's wrong, and not good.
So apply the PR, clear existing url keys (first run), reindex and clear cache (second run).

Check url keys generated:
- all store views: `massaznyj-apparat-dla-pohudenia-a`
- first store: `massazhnyy-apparat-dlya-pokhudeniya-b`
- second store: `massaznyj-apparat-dla-pohudenia-c`

That's better.

Extra: for CMS / Pages, when an user create a new page without url key, an url key is generated.

### Contribution checklist
